src/matrix_content_scanner/scanner/file_downloader.py

```diff
@@ -324,10 +349,19 @@ async def _get(
         try:
             logger.info("Sending GET request to %s", url)
             async with aiohttp.ClientSession() as session:
+                # TODO: Test we don't persist auth token
+                request_headers = copy.deepcopy(self._headers)
+                if auth_header is not None:
+                    auth_dict = {"Authorization": auth_header}
+                    if request_headers is None:
+                        request_headers = auth_dict
+                    else:
+                        request_headers.update(auth_dict)
```

Comment:
maybe this is simpler (and less footgunny since you never mutate anything, only construct), up to you though

```suggestion
                if auth_header is not None:
                     request_headers = {"Authorization": auth_header, **self._headers}
                else:
                     request_headers = self._headers                
```

Reply:
Ah! That's the python-foo I was looking for